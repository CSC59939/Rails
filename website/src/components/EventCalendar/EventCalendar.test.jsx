import React from 'react';
import renderer from 'react-test-renderer';
import {
  configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { EventCalendar } from '..';

configure({ adapter: new Adapter() });
const myEventsList = [
  {
    eventName: 'Project Presentation',
    title: 'CSC 59929',
    start: new Date('2018-12-04T21:42:00'),
    end: new Date('2018-12-04T23:45:00'),
    priority: 2,
  },
  {
    eventName: 'Lab Due',
    title: 'CSC 59928',
    start: new Date('2018-12-04T21:45:00'),
    end: new Date('2018-12-04T23:45:00'),
    priority: 1,
  },
];

// mock moment to return the same seperate times so we always render the same
// calendar range


describe('EventSummaryCollection', () => {
  it('Snapshot test of EventCalendar in the default agenda view', () => {
    let mockNCalls = 0;
    jest.mock('moment', () => () => ({
      format: () => {
        if (mockNCalls > 0) {
          return '2018-12-03T23:55:00';
        }
        mockNCalls += 1;
        return '2018-12-03T21:10:00';
      },
    }));
    expect(renderer.create(<EventCalendar events={myEventsList} />)
      .toJSON()).toMatchSnapshot();
  });
});
