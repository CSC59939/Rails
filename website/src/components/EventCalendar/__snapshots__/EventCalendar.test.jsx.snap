// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSummaryCollection Snapshot test of EventCalendar in the default agenda view 1`] = `
<div
  className="EventCalendarContainer"
>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Today
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        12/02/2018 â€” 01/01/2019
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className=""
          onClick={[Function]}
          type="button"
        >
          Month
        </button>
        <button
          className="rbc-active"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      className="rbc-agenda-view"
    >
      <span
        className="rbc-agenda-empty"
      >
        There are no events in this range.
      </span>
    </div>
  </div>
</div>
`;
